File: unique_values.py
def unique_values(arr):
    # Use a set to store unique values
    unique_set = set()
    unique_list = []

    for num in arr:
        if num not in unique_set:
            unique_set.add(num)
            unique_list.append(num)

    return unique_list

# Taking user input
user_input = input("Enter a list of integers separated by spaces: ")
input_array = list(map(int, user_input.split()))

# Getting the unique values
result = unique_values(input_array)
print(f"Unique values: {result}")

File: unique_values.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function uniqueValues(arr) {
    // Use a Set to store unique values
    const uniqueSet = new Set();
    const uniqueArray = [];

    for (const num of arr) {
        if (!uniqueSet.has(num)) {
            uniqueSet.add(num);
            uniqueArray.push(num);
        }
    }

    return uniqueArray;
}

// Taking user input
rl.question("Enter a list of integers separated by spaces: ", function(userInput) {
    const inputArray = userInput.split(' ').map(Number);

    // Getting the unique values
    const result = uniqueValues(inputArray);
    console.log(`Unique values: ${result}`);

    rl.close();
});
